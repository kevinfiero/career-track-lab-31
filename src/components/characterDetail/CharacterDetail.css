section{
  width: 100%;
  display: flex;
  flex-direction: column;
}

/* img{
  width: 18rem;
  height: 16rem;
} */ 

figcaption{
  width: 22rem;
}

figure{
  text-align: center;
  align-items: center;
}

div{
  display: flex;
  flex-direction: row;
}

h1, label{
  font-size: 1rem;
  margin: 1rem;
}

label{
  width: 5rem;
}
