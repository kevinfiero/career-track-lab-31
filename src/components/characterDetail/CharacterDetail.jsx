import React from 'react';
import PropTypes from 'prop-types';
import './CharacterDetail.css';

const CharacterDetail = ({ image, name }) => (
  <section>
    <figure>
      <img src={image} />
      <figcaption>
        <h1>{name}</h1>
      </figcaption>
    </figure>
  </section>
);

CharacterDetail.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default CharacterDetail;
