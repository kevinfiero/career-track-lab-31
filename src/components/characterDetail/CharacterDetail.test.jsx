import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetail', () => {
    const { asFragment } = render(
      <CharacterDetail 
        image="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
        name="Arnold Shortman"
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
