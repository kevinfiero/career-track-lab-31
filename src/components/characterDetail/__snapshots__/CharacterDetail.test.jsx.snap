// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <section>
    <figure>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
      />
      <figcaption>
        <h1>
          Arnold Shortman
        </h1>
      </figcaption>
    </figure>
  </section>
</DocumentFragment>
`;
